src/nfa/delta.cc

```diff
@@ -512,3 +526,169 @@ Delta::Transitions::const_iterator Delta::Transitions::end() const { return cons
 StatePost::Moves::Moves(
     const StatePost& state_post, StatePost::const_iterator symbol_post_it, StatePost::const_iterator symbol_post_end)
     : state_post_{ &state_post }, symbol_post_it_{ symbol_post_it }, symbol_post_end_{ symbol_post_end } {}
+
+void Delta::add_symbols_to(OnTheFlyAlphabet& target_alphabet) const {
+    size_t aut_num_of_states{ num_of_states() };
+    for (mata::nfa::State state{ 0 }; state < aut_num_of_states; ++state) {
+        for (const SymbolPost& move: state_post(state)) {
+            target_alphabet.update_next_symbol_value(move.symbol);
+            target_alphabet.try_add_new_symbol(std::to_string(move.symbol), move.symbol);
+        }
+    }
+}
+
+OrdVector<Symbol> Delta::get_used_symbols() const {
+    //TODO: look at the variants in profiling (there are tests in tests-nfa-profiling.cc),
+    // for instance figure out why NumberPredicate and OrdVedctor are slow,
+    // try also with _STATIC_DATA_STRUCTURES_, it changes things.
+
+    //below are different variant, with different data structures for accumulating symbols,
+    //that then must be converted to an OrdVector
+    //measured are times with "mata::nfa::get_used_symbols speed, harder", "[.profiling]" now on line 104 of nfa-profiling.cc
+
+    //WITH VECTOR (4.434 s)
+    //return get_used_symbols_vec();
+
+    //WITH SET (26.5 s)
+    //auto from_set = get_used_symbols_set();
+    //return utils::OrdVector<Symbol> (from_set .begin(),from_set.end());
+
+    //WITH NUMBER PREDICATE (4.857s) (NP removed)
+    //return utils::OrdVector(get_used_symbols_np().get_elements());
+
+    //WITH SPARSE SET (haven't tried)
+    //return utils::OrdVector<State>(get_used_symbols_sps());
+
+    //WITH BOOL VECTOR (error !!!!!!!):
+    //return utils::OrdVector<Symbol>(utils::NumberPredicate<Symbol>(get_used_symbols_bv()));
+
+    //WITH BOOL VECTOR (1.9s):
+    std::vector<bool> bv{ get_used_symbols_bv() };
+    utils::OrdVector<Symbol> ov{};
+    const size_t bv_size{ bv.size() };
+    for (Symbol i{ 0 }; i < bv_size; ++i) { if (bv[i]) { ov.push_back(i); } }
+    return ov;
+
+    ///WITH BOOL VECTOR, DIFFERENT VARIANT? (1.9s):
+    //std::vector<bool> bv = get_used_symbols_bv();
+    //std::vector<Symbol> v(std::count(bv.begin(), bv.end(), true));
+    //return utils::OrdVector<Symbol>(v);
+
+    //WITH CHAR VECTOR (should be the fastest, haven't tried in this branch):
+    //BEWARE: failing in one noodlificatoin test ("Simple automata -- epsilon result") ... strange
+    //BoolVector chv = get_used_symbols_chv();
+    //utils::OrdVector<Symbol> ov;
+    //for(Symbol i = 0;i<chv.size();i++)
+    //    if (chv[i]) {
+    //        ov.push_back(i);
+    //    }
+    //return ov;
+}
+
+// Other versions, maybe an interesting experiment with speed of data structures.
+// Returns symbols appearing in Delta, pushes back to vector and then sorts
+mata::utils::OrdVector<Symbol> Delta::get_used_symbols_vec() const {
+#ifdef _STATIC_STRUCTURES_
```

Comment:
we should get rid of this stuff somehow

Reply:
When one perf tests the used symbols variations, I will gladly remove all the unused versions.